File: src/main/java/nl/jhvh/java/geometry/GeometryUtil.java
package nl.jhvh.java.geometry;

import org.apache.commons.lang3.NotImplementedException;

public class GeometryUtil {

    private GeometryUtil() {
        throw new NotImplementedException(this.getClass().getName() + " is a utility class; do not instantiate!");
    }

    public static final double RECTANGLE_DEGREES = 90;

    public static double radiansToDegreesFactor = 180 / Math.PI;
    public static double degreesToRadiansFactor = Math.PI / 180;

    public static double radiansToDegrees(double radians) {
        return radians * radiansToDegreesFactor;
    }

    public static double getDegreesToRadians(double radians) {
        return radians * degreesToRadiansFactor;
    }

}

File: src/main/kotlin/nl/jhvh/kotlin/geometry/GeometryUtil.kt
/**
 * This is for DEMO only to show difference between Java and Kotlin (using extension methods in Kotlin).
 * For any real use, make sure to use the stuff from the `javax.measure` package instead !!
 */

package nl.jhvh.kotlin.geometry

const val rectangleDegrees: Double = 90.0

const val radiansToDegreesFactor: Double = 180 / Math.PI
const val degreesToRadiansFactor: Double = Math.PI / 180

fun Double.radiansToDegrees() = this * radiansToDegreesFactor
fun Double.degreesToRadians() = this * degreesToRadiansFactor
